### PR TITLE
Feature/switch to avfoundation

### PR DESCRIPTION
**Pull request by rob5408**

### CODE REVIEW

File: Pod/Classes/MWPhotoBrowser.m [hunk 1]
```diff
@@ -1295,10 +1307,17 @@ - (void)videoFinishedCallback:(NSNotification*)notification {
...
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:AVPlayerItemFailedToPlayToEndTimeNotification
+                                                  object:_currentVideoPlayerItem];
     
     // Clear up
-    [self clearCurrentVideo];
-    [self dismissViewControllerAnimated:YES completion:nil];
+    __typeof(self) __weak weakSelf = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        __typeof(self) strongSelf = weakSelf;
```
Comment (rickbdotcom) [inline Pod/Classes/MWPhotoBrowser.m]: Any particular reason for using strongSelf here instead of weakSelf?

Comment (rickbdotcom) [inline Pod/Classes/MWPhotoBrowser.m]: actually i'm not even sure why we need to use weak self here for that matter.

File: Pod/Classes/MWPhotoBrowser.m [hunk 2]
```diff
@@ -1282,6 +1286,14 @@ - (void)_playVideo:(NSURL *)videoURL atPhotoIndex:(NSUInteger)index {
                                                  name:AVPlayerItemDidPlayToEndTimeNotification
                                                object:_currentVideoPlayerItem];
 
+    [[NSNotificationCenter defaultCenter] removeObserver:self
```
Comment (rickbdotcom) [inline Pod/Classes/MWPhotoBrowser.m]: I don't quite follow the logic here why we're removing the observer then adding it back.

File: Pod/Classes/MWPhotoBrowser.m (hunk 1 above)
Comment (rob5408) [inline Pod/Classes/MWPhotoBrowser.m]: Yeah, I didn't even have it originally. Then started to overthink it. I can take it out.

File: Pod/Classes/MWPhotoBrowser.m (hunk 2 above)
Comment (rob5408) [inline Pod/Classes/MWPhotoBrowser.m]: I was following the pattern they had in place before. That said I have been the victim of receiving double notifications before.

**Pull request merged**